src/app/content/highlights/hooks/openMyHighlights.ts

```diff
@@ -1,17 +1,29 @@
 import { user } from '../../../auth/selectors';
+import { push } from '../../../navigation/actions';
+import { query } from '../../../navigation/selectors';
+import { getQueryForParam } from '../../../navigation/utils';
 import { ActionHookBody } from '../../../types';
 import { actionHook } from '../../../utils';
+import { modalQueryParameterName } from '../../constants';
+import { getContentParams } from '../../utils/urlUtils';
 import { initializeMyHighlightsSummary, openMyHighlights } from '../actions';
+import { modalUrlName } from '../constants';
 import * as select from '../selectors';
 
 export const hookBody: ActionHookBody<typeof openMyHighlights> = ({
   dispatch, getState,
 }) => async() => {
   const state = getState();
   const authenticated = user(state);
+  const existingQuery = query(state);
+
   const summaryNeedsInitialization = () => select.summaryHighlights(state) === null
     && select.summaryIsLoading(state) === false;
 
+  dispatch(push(getContentParams(state), {
```

Comment:
same idea, i believe you can get all your necessary information out of the navigation state, and could create a canned hook just for opening dialogs. you can have multiple hooks on a given action, or you can compose them manually if you need them to happen in a certain order. this might look something like

```
export const initializeMyHighlightsHook = actionHook(openMyHighlights, hookBody);
export const openMyHighlightsHook = actionHook(openMyHighlights, openModalHook(modalUrlName));
```

Reply:
I need the location state too since based on it highlights are cleared on location change. And since the location state is missing when navigated to a page via a link, I'm trying to generate it as normal links do.